docker/Dockerfile

```diff
@@ -281,17 +282,31 @@ RUN set -e; \
     git clone https://github.com/xkszltl/roaster.git /etc/roaster/scripts; \
     truncate -s0 ~/.bash_history;
 
+# [PEP 668](https://peps.python.org/pep-0668/) disallows installing pip packages directly to the base environment.
+# However, there are no viable alternatives, so we set `--break-system-package` as a workaround.
 RUN set -e; \
+    PIP_MAYBE_BREAK_SYSTEM_PACKAGES="$(set -e; \
+        OPT="--break-system-packages"; \
+        python3 -m pip install --help \
+        | sed 's/^/ /' \
+        | sed 's/$/ /' \
+        | grep -q "[^[:alnum:]_-]$OPT[^[:alnum:]_-]" \
```

Comment:
`$(printf '%s' "$OPT" | sed 's/\([\.\-]\)/\\\1/g')` to escape

Reply:
And - should also be escaped even it is at the end.

Reply:
Usually I would just sed it out and head -n1, instead of && printf, this way you don't need have have a var because you only need to specify it once.

Reply:
Try to keep this a one-liner because we'll need to put it in many places in-line.

Reply:
Something like
```
python3 -m pip install --help | sed 's/^/ /' | sed 's/$/ /' | grep "[^[:alnum:]_\-]$(printf '%s' '--break-system-packages' | sed 's/\([\.\-]\)/\\\1/g')[^[:alnum:]_\-]" | sed 's/^[^-]*\-\-/\-\-/' | sed 's/[^[:alnum:]_\-].*$//'
```
?

No, that looks too ugly for me and we will probably not be able to reuse this type of one-liner. If you come up with anything better feel free to change it as a follow-up.

Reply:
`| sed 's/\(.*\)/ \1 /' | sed -n 's/.*[^[:alnum:]_\-]\(\-\-break\-system\-package\)[^[:alnum:]_\-].*/\1/p' | head -n1`